src/sources/prometheus/parser.rs

```diff
@@ -15,11 +15,13 @@ fn has_values_or_none(tags: BTreeMap<String, String>) -> Option<BTreeMap<String,
     }
 }
 
-fn utc_timestamp(timestamp: Option<i64>) -> Option<DateTime<Utc>> {
-    timestamp.and_then(|timestamp| {
-        Utc.timestamp_opt(timestamp / 1000, (timestamp % 1000) as u32 * 1000000)
-            .latest()
-    })
+fn utc_timestamp(timestamp: Option<i64>, now: &mut Option<DateTime<Utc>>) -> DateTime<Utc> {
```

Comment:
The API here seems a bit odd to me. I might just always create `now` on every scrape and remove the `Option<>` here. I realize what you have here is lazier and won't fetch now unless it is needed, but I think it'd be more straight-forward. Also, creating `now` closer to the collection might be more accurate not that I expect initializing it later is likely to add a noticeable delay.